homeassistant/components/homekit/type_thermostats.py

```diff
@@ -76,26 +77,37 @@
 HC_HOMEKIT_VALID_MODES_WATER_HEATER = {"Heat": 1}
 UNIT_HASS_TO_HOMEKIT = {TEMP_CELSIUS: 0, TEMP_FAHRENHEIT: 1}
 
+HC_HEAT_COOL_OFF = 0
+HC_HEAT_COOL_HEAT = 1
+HC_HEAT_COOL_COOL = 2
+HC_HEAT_COOL_AUTO = 3
+
+HC_MIN_TEMP = 10
+HC_MAX_TEMP = 38
+
 UNIT_HOMEKIT_TO_HASS = {c: s for s, c in UNIT_HASS_TO_HOMEKIT.items()}
 HC_HASS_TO_HOMEKIT = {
-    HVAC_MODE_OFF: 0,
-    HVAC_MODE_HEAT: 1,
-    HVAC_MODE_COOL: 2,
-    HVAC_MODE_AUTO: 3,
-    HVAC_MODE_HEAT_COOL: 3,
-    HVAC_MODE_FAN_ONLY: 2,
+    HVAC_MODE_OFF: HC_HEAT_COOL_OFF,
+    HVAC_MODE_HEAT: HC_HEAT_COOL_HEAT,
+    HVAC_MODE_COOL: HC_HEAT_COOL_COOL,
+    HVAC_MODE_AUTO: HC_HEAT_COOL_AUTO,
+    HVAC_MODE_HEAT_COOL: HC_HEAT_COOL_AUTO,
+    HVAC_MODE_DRY: HC_HEAT_COOL_COOL,
+    HVAC_MODE_FAN_ONLY: HC_HEAT_COOL_COOL,
 }
 HC_HOMEKIT_TO_HASS = {c: s for s, c in HC_HASS_TO_HOMEKIT.items()}
```

Comment:
This is still a problem as there are multiple `HC_HEAT_COOL_COOL` in `HC_HASS_TO_HOMEKIT `

Reply:
https://github.com/home-assistant/core/pull/34073/files#diff-20f95faa328f4cba2dd52175cb9242baL160

Only line 160 we build a map out of this with the valid values

Reply:
I'm going to move discussion on the `daikin` issue back to that case as its not related to this change.